test/test_defsfont_preset_iteration.c

```diff
@@ -0,0 +1,40 @@
+#include "test.h"
+#include "fluidsynth.h"
+#include "sfloader/fluid_sfont.h"
+#include "utils/fluidsynth_priv.h"
+
+int main(void)
+{
+    char *s;
+    int id;
+    fluid_sfont_t *sfont;
+    fluid_preset_t *preset;
+    fluid_preset_t *prev_preset;
+    int count = 0;
+
+    /* setup */
+    fluid_settings_t *settings = new_fluid_settings();
+    fluid_synth_t *synth = new_fluid_synth(settings);
+    fluid_settings_dupstr(settings, "synth.default-soundfont", &s);
+    id = fluid_synth_sfload(synth, s, 1);
+    sfont = fluid_synth_get_sfont_by_id(synth, id);
+
+    /* code under test */
+    fluid_sfont_iteration_start(sfont);
+
+    while ((preset = fluid_sfont_iteration_next(sfont)) != NULL) {
+        count++;
+
+        /* make sure we actually got a different preset */
+        TEST_ASSERT(preset != prev_preset);
+        prev_preset = preset;
+    }
+
+    TEST_ASSERT(count > 0);
```

Comment:
Yes, I though so as well. But in that case I think it would be good to have the VintageDreams visible as a dependency in the test. Simply loading the DEFAULT_SOUNDFONT and then testing against 136 is too magic and not explicit enough, IMHO.

Reply:
And I'm not a big fan of all these different TEST_ASSERT macros. I think TEST_ASSERT (or maybe just simply ASSERT?) with a boolean expression is much more readable. In UnitTest style frameworks those different asserts do make at least a little sense, because they output different errors depending on the assertion type.  But here we don't make that distinction.